src/libs/fileDownload/FileUtils.ts

```diff
@@ -285,6 +285,15 @@ function verifyFileFormat({fileUri, formatSignatures}: {fileUri: string; formatS
         });
 }
 
+function isOfflineFile(receiptUri?: string | number): boolean {
+    if (!receiptUri) {
+        return false;
+    }
+    return (
+        typeof receiptUri === 'number' || receiptUri?.startsWith('blob:') || receiptUri?.startsWith('file:') || (receiptUri?.startsWith('/') && !receiptUri?.startsWith('/chat-attachments'))
```

Comment:
> @Kalydosos can you clarify a bit why we need this change here? Thanks

@hoangzinh already loaded images before going offline have an adress that starts with "/chat-attachments'" and with the previously used function  
https://github.com/Kalydosos/App/blob/92948a59b9a9b49ba6c483bf29d74dd194b18775/src/libs/fileDownload/FileUtils.ts#L297-L302
they were seen as "local" files and when offline, the offline view was not displayed when clicked on the images. The new function was used here  https://github.com/Kalydosos/App/blob/92948a59b9a9b49ba6c483bf29d74dd194b18775/src/components/ImageView/index.tsx#L238-L239  . I could not reused https://github.com/Kalydosos/App/blob/92948a59b9a9b49ba6c483bf29d74dd194b18775/src/libs/fileDownload/FileUtils.ts#L297-L302  in the new function, it was safer to copy and adapt it

Reply:
maybe the behaviour changes at sometime but i could not reproduce the "before changes" of your video  https://github.com/Expensify/App/pull/50511#issuecomment-2426341037 unless i apply that modification but i was played by my browser cache in the beginning, but once disabled the cache i could not reproduce the "before changes" unless  i made that modification. On whch version did you made the video ?

Reply:
> I couldn't find previous version

i mean a version or a tag of the app where i can reproduce that : clicking on a well loaded attachment before going offline will display "offliex view" once clicked and offline

Reply:
> > already loaded images before going offline have an adress that starts with "/chat-attachments'"
> 
> I think it's not correct @Kalydosos. If the attachment comes from chat report, it will have start with `/chat-attachments`. For example: `/chat-attachments/9208320899643172368/w_cf3e867bf4b45d8dcc5d8168166247b2b44e8a8f.png`
> 
> Therefore, the function name `isOfflineFile` does not really make sense. I think we can either update the existing "isLocalFile" function or add a new function to check `reportAttachmentFile` and use combine it with "isLocalFile". What do you think?

the function was named like that beacuse it only returns true for files loaded when offline in fact. To match this scenario

if the image is loaded when offline and clicked upon when offline => image displayed

>  we can either update the existing "isLocalFile" function or add a new function to check `reportAttachmentFile` and use combine it with "isLocalFile".

i will see for this once i grasp the changes you are talking about in the comments below.

Reply:
> I see. Do you think it's an existing bug in the main branch?

@hoangzinh i have done the test on the main branch and it turns out that the situation below is the same once you disable the browser cache (btw your cache was enable during your test https://github.com/Expensify/App/pull/50511#issuecomment-2426341037 that's why the problem didnt occur)

> already loaded images before going offline have an adress that starts with "/chat-attachments'" and with the previously used function
https://github.com/Kalydosos/App/blob/92948a59b9a9b49ba6c483bf29d74dd194b18775/src/libs/fileDownload/FileUtils.ts#L297-L302
they were seen as "local" files and when offline, the offline view was not displayed when clicked on the images.

our PR will fix that issue by the same time

Reply:
> I think it's not correct @Kalydosos. If the attachment comes from chat report, it will have start with `/chat-attachments`. For example: `/chat-attachments/9208320899643172368/w_cf3e867bf4b45d8dcc5d8168166247b2b44e8a8f.png`

yes i speaking of report attachment in the context when the user add them in a chat thread as specify by the ticket https://github.com/Expensify/App/issues/49974

Reply:
> Therefore, the function name isOfflineFile does not really make sense.

i propose to name it isFileLocalToUserDevice because that the only context when it should return true.

> we can either update the existing "isLocalFile" function or add a new function to check reportAttachmentFile and use combine it with "isLocalFile"

the function isLocalFile serves it purpose, it indicates whether a file is local generally speaking base on the structure of its uri, and that function is used by many other components, so i think we should change it. For adding a new function, we only have the uri and its a simple  basic check on its structure. I advise we keep the "isOfflineFile" and hjust rename it isFileLocalToUserDevice

Reply:
> const isNotAttachmentFile = !url?.startsWith('/chat-attachments');

@hoangzinh  typeof url can be number sometimes, so this part above could bug i think

so maybe add a check on that part ? otherwise it's a good idea to make the change inside the file and i propose we rename the variable const isNotRemoteAttachmentFile (because it is an attachment file after all but local to the user device not on the server yet because the user is offline)

but it's a good idea :+1:

Reply:
@hoangzinh i will also try to come up with a proposition along that idea :+1:

Reply:
@hoangzinh  i propose this by taking inspiration from your advice


> // isLocalToUserDeviceFile means the file is located on the user device,
    // not loaded on the server yet (the user is offline when loading this file in fact)
    let isLocalToUserDeviceFile = FileUtils.isLocalFile(url);
    if (isLocalToUserDeviceFile && (typeof url === "string") && url.startsWith("/chat-attachments")){
        isLocalToUserDeviceFile = false ;
    }

i included some comments to make the variable meaning clearer

I tested and it works well. Any thought ?